src/main/java/io/lakefs/routerfs/PathMapper.java

```diff
@@ -1,7 +1,12 @@
 package io.lakefs.routerfs;
 
+import io.lakefs.routerfs.dto.DefaultSchemeTranslation;
+import io.lakefs.routerfs.dto.PathProperties;
+import lombok.Builder;
 import lombok.Getter;
+import lombok.NonNull;
```

Comment:
Not a fan of `NonNull` annotations, except if they are enforced at compile time.  Can we use some appropriate [Java checker](https://checkerframework.org/manual/#nullness-checker) instead?  I somehow recall it is a [part of Java 8](https://blogs.oracle.com/java/post/java-8s-new-type-annotations).